dagger/tasks/task.py

```diff
@@ -1311,6 +1315,23 @@ class DefaultTemplateDAGInstance(ITemplateDAGInstance[str, str]):
     Default Implementation of ITemplateDAGInstance
     """
 
+    async def stop(self) -> None:
+        remaining_tasks: Optional[List[ITask]] = await self.get_remaining_tasks(
+            next_dag_id=self.root_dag, workflow_instance=self, tasks=[]  # type: ignore
+        )  # type: ignore
+        if remaining_tasks:
+            for task in remaining_tasks:
+                if task.status.code == TaskStatusEnum.EXECUTING.name:
+                    await task.stop()
+                if task.status.code not in TERMINAL_STATUSES:
```

Comment:
Just so I understand:
- Executing tasks need to have super.stop() called
- Not started / other non-terminal statuses do not need super.stop() / only need to have their status updated to STOPPED
- Terminal statuses are left as is

Reply:
correct!